app/graphql/types/query_type.rb

```diff
@@ -191,5 +189,64 @@ def query_files_whitelist(path, file_type)
       def query_file?(file, path, file_type)
         File.exist?(file) && query_files_whitelist(path, file_type).include?(file)
       end
+
+      def static_content_with_data_type(data_type, name, version = nil)
+        # HINT: mysql REGEXP for /^#{name}(-\d\.\d\.\d)?$/ was not working fully somehow,
+        #       so we query for all contents starting with name and filter with rails afterwards
+        static_contents = StaticContent.where("data_type = '#{data_type}' AND name LIKE '#{name}%'")
```

Comment:
@marcometz · 4 weeks ago

> wenn ich 2 Statics habe ("Haus" und "Haustier") und dann die Methode Aufrufe mit "Haus", dann finde ich beide, oder? Erst durch die Stunts unten werden dann wieder "falsche" rausgeworfen.
> 
> Ich glaube ich würde eher das Datenmodel von StaticContetent erweitern um ein Attribut :version_number und dann in der Abfrage explizit darauf filtern, bzw danach sortieren wenn es keine konkrete Versionsnummer gibt und die letzte/erste zu erhalten:
> 
> static_contents = StaticContent.where(data_type: data_type, name: name).order(:version_number).last if version.blank? static_contents = StaticContent.where(data_type: data_type, name: name, version_number: version) if version.present?

Reply:
@digorath do you see some downsides of having a version number field instead of appended in the name string of a static content? what differences does that solution bring up on the mobile app part? none, right? it would still request for a name and a version. so all changes would happen here around getting the right content i guess.
@marcometz do you think it is worth the effort now or should be postpone a refactoring at a later point, where we have more time? do you see some critical places currently, which we cannot roll out for a first version?

Reply:
if we have the version number as an additional field like name and content, it should work the same from the perspective of the app. nothing would need to change there.

if that got changed on the server, we would need to migrate all the versioned static contents that are versioned through the name suffix.
This is currently not a problem yet, but if we delay it, then we might miss some static contents somewhere. so I suggest doing it now, if you want to do it.

I do think it is a better solution than the current one.

Reply:
I am on it. The quick code example of @marcometz does not work in all cases. If there is a version number given, we do not only want to return exactly that one but the closest matching version.

Reply:
if have added a `version` field and translated the code to use this one instead of regex on names, see 167d856